### PR TITLE
B 2

### PR DESCRIPTION
**Pull request by sergey-gudimov**
b-2

### CODE REVIEW

File: src/main/java/com/gudimov/movieland/controller/MovieController.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.gudimov.movieland.controller;
 
 import com.gudimov.movieland.entity.Movie;
+import com.gudimov.movieland.entity.MovieWithGenreCountry;
```
Comment (juliamanunia) [inline src/main/java/com/gudimov/movieland/controller/MovieController.java]: Why do we need a separate entity for this?

File: src/main/java/com/gudimov/movieland/controller/MovieController.java [hunk 2]
```diff
@@ -39,5 +40,20 @@
             return new ResponseEntity<>(movieJson, HttpStatus.OK);
         }
     }
+
+    @RequestMapping(value = "/random", method = RequestMethod.GET, produces = "application/json;charset=UTF-8")
```
Comment (juliamanunia) [inline src/main/java/com/gudimov/movieland/controller/MovieController.java]: produces and method can be applied in @RequestMapping in class definition

File: src/main/java/com/gudimov/movieland/dao/jdbc/MovieDaoJdbc.java [hunk 3]
```diff
@@ -27,11 +25,28 @@
...
         return listMovie;
     }
+
+    @Override
+    public List<MovieWithGenreCountry> getRandom() {
+        log.info("Start query to get random movie from DB");
+        List<Movie> listMovie = jdbcTemplate.query(getRandomMovieSQL, new MovieRowMapper());
+        log.info("Finish query to get movie from DB");
+        List<MovieWithGenreCountry> movieWithGenreCountries;
+        movieWithGenreCountries=MovieEnricher.EnrichMovieWithGenre(listMovie);
```
Comment (juliamanunia) [inline src/main/java/com/gudimov/movieland/dao/jdbc/MovieDaoJdbc.java]: don't separate variable declaration and it's assignment. You should enrich Movie by Genre and Country in MovieService, but not in Jdbc. Also method name should start from lower case letter

File: src/main/java/com/gudimov/movieland/controller/MovieController.java (hunk 1 above)
Comment (sergey-gudimov) [reply src/main/java/com/gudimov/movieland/controller/MovieController.java]: Class MovieWithGenreCountry extends class Movie

File: src/test/java/com/gudimov/movieland/controller/MovieControllerTest.java [hunk 4]
```diff
@@ -85,4 +87,95 @@ public void getAllMovie() throws Exception {
...
+        genre1.setId(1);
+        genre1.setName("драма");
+        genre2 = new Genre();
+        genre2.setId(6);
+        genre2.setName("мелодрама");
+        movieSecond.setGenre(Arrays.asList(genre1, genre2));
+
+        when(movieServiceMock.getAll()).thenReturn(Arrays.asList(movieFirst, movieSecond));
+
+        mockMvc.perform(get("/v1/movie")).andExpect(status().isOk())
```
Comment (juliamanunia) [inline src/test/java/com/gudimov/movieland/controller/MovieControllerTest.java]: incorrect path, should be /movie/random. Please put v1 into DispatcherServlet url

File: src/main/java/com/gudimov/movieland/controller/MovieController.java (hunk 2 above)
Comment (sergey-gudimov) [reply src/main/java/com/gudimov/movieland/controller/MovieController.java]: Fixed

File: src/test/java/com/gudimov/movieland/controller/MovieControllerTest.java (hunk 4 above)
Comment (sergey-gudimov) [reply src/test/java/com/gudimov/movieland/controller/MovieControllerTest.java]: Fixed

File: src/main/java/com/gudimov/movieland/dao/jdbc/MovieDaoJdbc.java (hunk 3 above)
Comment (sergey-gudimov) [reply src/main/java/com/gudimov/movieland/dao/jdbc/MovieDaoJdbc.java]: Fixed

**Pull request merged**